codalab/worker/singularity_image_manager.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import os
+import threading
+
+from codalab.lib.formatting import size_str
+from codalab.worker.image_manager import ImageAvailabilityState
+from codalab.worker.fsm import DependencyStage
+from codalab.worker.image_manager import ImageManager
+from codalab.worker.singularity_utils import get_singularity_container_size
+from spython.main import Client  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class SingularityImageManager(ImageManager):
+    def __init__(self, max_image_size: int, max_image_cache_size: int, image_folder: str):
+        """
+        image_folder needs to be an absolute path
```

Comment:
Make the docstrings match the convention (https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)